### ISSUE TITLE
make svd and kmeans like rest of tests (for make test and make testperf)

### ISSUE THREAD

**Issue by pseudotensor** [h2oai/h2o4gpu#307]
E.g. Currently test_tsvd_bench.py is done during all tests and (e.g.) kills my home system with only 32GB of system memory because it also uses sklearn.  For tests we should only test h2o4gpu and not also sklearn (unless there is an assert on accuracy, which there is none in test_tsvd_bench.py).

Can you make your svd tests like the rest of our tests so that "make test" will only check h2o4gpu and "make testperf" will show performance stuff?  Nvidia want to track the performance of h2o4gpu over time and our "make testperf" has the desired output for glm and xgboost but not svd or kmeans.

For now I'll skip those tsvd_bench tests when the CHECKPERFORMANCE environment variable is set.  BTW, that flag was mangled and testperf wasn't working correctly before.

**Comment by pseudotensor** [comment]
@mdymczyk Randomly got the below.  We shouldn't have performance tests in the normal jenkins build because we aren't ensured we are running on a clean unused system.  Please move as a testperf job.

}
            for arg in pyfuncitem._fixtureinfo.argnames:
                testargs[arg] = funcargs[arg]
>           testfunction(**testargs)

/root/.pyenv/versions/3.6.1/lib/python3.6/site-packages/_pytest/python.py:147: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_kmeans.TestKmeans object at 0x7fcd5409ceb8>

    def test_speed_vs_sk(self):
        from sklearn.cluster import KMeans as skKMeans
        n_samples = 100000
        centers = 10
        X, true_labels = make_blobs(n_samples=n_samples, centers=centers,
                                    cluster_std=1., random_state=42)
    
        kmeans_h2o = KMeans(n_gpus=1, n_clusters=centers, random_state=42)
        # Warmup - during first call CUDA kernels take ~2sec to load
        kmeans_h2o.fit(X)
        start_h2o = time.time()
        kmeans_h2o.fit(X)
        end_h2o = time.time()
    
        kmeans_sk = skKMeans(n_init=1, n_clusters=centers, algorithm='full', n_jobs=-1)
        start_sk = time.time()
        kmeans_sk.fit(X)
        end_sk = time.time()
    
>       assert end_h2o - start_h2o <= end_sk - start_sk
E       assert (1512288877.9139025 - 1512288876.5533953) <= (1512288879.1286077 - 1512288877.9139316)

tests_open/test_kmeans.py:169: AssertionError
===================== 1 failed, 8 passed in 14.63 seconds ======================
Makefile:516: recipe for target 'dotestfast' failed
make: *** [dotestfast] Error 1
